### PR TITLE
POST oAuth2Permissiongrants - new doc

### PR DESCRIPTION
**Pull request by bpcastilho**
POST was missing in the docs.

### CODE REVIEW

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+doc_type: apiPageType
+ms.prod: ""
+author: ""
+---
+
+# Create oAuth2PermissionGrant
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Createte the properties of oAuth2PermissionGrant object.
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: Typo

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account) | Directory.ReadWrite.All, Directory.AccessAsUser.All    |
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: OK - we have some newer permissions here that can be used.  Also I don't think Directory.RW.All allows this.

DelegatedPermissionGrant.ReadWrite.All is the one you want.  Put that in both Delegated and Application rows.  You can keep the AccessAsUser permission.

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+|Delegated (personal Microsoft account) | Not supported.    |
+|Application | Directory.ReadWrite.All |
+
+## HTTP request
+<!-- { "blockType": "ignored" } -->
+```http
+POST /oAuth2Permissiongrants/{id}
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: POST /oauth2Permissiongrants

Remove the other 2

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+POST /users/{id | userPrincipalName}/oAuth2Permissiongrants/{id}
+POST /drive/root/createdByUser/oAuth2Permissiongrants/{id}
+```
+## Request headers
+| Name       | Type | Description|
+|:-----------|:------|:----------|
+| Authorization  | string  | Bearer {token}. Required. |
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: The auto-generation tool does generate the right info here.
Add another row to the table
| Content-Type | application/json |

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+## Request body
+In the request body, supply a JSON representation of [oAuth2PermissionGrant](oauth2permissiongrant.md) object.
+
+| Property	   | Type	|Description|
+|:---------------|:--------|:----------|
+|scope|String| Specifies the value of the scope claim that the resource application should expect in the OAuth 2.0 access token. |
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: OK the required fields are
clientId, consentType, principalId, resourceId and scope.

Try creating an oauth2PermissionGrant in Graph Explorer you can use that to determine the required properties

**Comment by bpcastilho** [comment]
Closing this and starting new PR.

**Pull request closed without merging**